### ISSUE TITLE
Cannot load default font (RuntimeError)

### ISSUE THREAD

**Issue by Micka33** [gosu/gosu#384]
```ruby
require 'gosu'

module ZOrder
  BACKGROUND, STARS, PLAYER, UI = *0..3;
end
class Main < Gosu::Window
  def initialize
    super 640, 480
    self.caption = "Tutorial Game"
    @font = Gosu::Font.new(20)
  end
  def draw
    @font.draw('Score: ', 10, 10, ZOrder::UI, 1.0, 1.0, Gosu::Color::YELLOW)
  end
end

Main.new.show
```

the console says 
```
 ruby ./test.rb
./test.rb:13:in `draw': Cannot load default font (RuntimeError)
	from ./test.rb:13:in `draw'
	from /Users/mickael/.rvm/gems/ruby-2.4.0/gems/gosu-0.11.2/lib/gosu/patches.rb:152:in `tick'
	from /Users/mickael/.rvm/gems/ruby-2.4.0/gems/gosu-0.11.2/lib/gosu/patches.rb:152:in `tick'
	from ./test.rb:17:in `<main>'
```

My Ruby version
```
system:
    uname:        "Darwin XXXXX.local 14.5.0 Darwin Kernel Version 14.5.0: Thu Jun 16 19:58:21 PDT 2016; root:xnu-2782.50.4~1/RELEASE_X86_64 x86_64"
    name:         "OSX"
    version:      "10.10"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)"
    zsh:          "/usr/local/bin/zsh => zsh 5.3.1 (x86_64-apple-darwin14.5.0)"
    remote_path:  "osx/10.10/x86_64"

  rvm:
    version:      "rvm 1.29.1 (master) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]"
    updated:      "59 minutes 48 seconds ago"
    path:         "/Users/mickael/.rvm"
    autolibs:     "[homebrew] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  ruby:
    interpreter:  "ruby"
    version:      "2.4.0p0"
    date:         "2016-12-24"
    platform:     "x86_64-darwin14"
    patchlevel:   "2016-12-24 revision 57164"
    full_version: "ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin14]"
```

**Comment by shawn42** [comment]
This code runs fine for me:

- NOT using RVM
- Ruby version 2.3.3p222 via rbenv
- Darwin byte-me.local 16.4.0 Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; 
- sdl2: stable 2.0.5 from homebrew

Just throwing this out there as another data point.

**Comment by Micka33** [comment]
@shawn42 
Can you tell me where the default font is supposed to be found ?

```
 brew info sdl2
sdl2: stable 2.0.5 (bottled), HEAD
Low-level access to audio, keyboard, mouse, joystick, and graphics
https://www.libsdl.org/
/usr/local/Cellar/sdl2/2.0.5 (84 files, 3.9MB) *
  Poured from bottle on 2017-03-20 at 19:45:41
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/sdl2.rb
```

**Comment by pmer** [comment]
The default font is supposed to be Arial.

**Comment by Micka33** [comment]
I tested the same code, with the following : 
```ruby
#@font = Gosu::Font.new(20, {name:'assets/fonts/Lato/Lato-Light.ttf'})
#@font = Gosu::Font.new(self, 'assets/fonts/Lato/Lato-Light.ttf', 20)
#@font = Gosu::Font.new(self, 'assets/fonts/Roboto/Roboto-Thin.ttf', 20)
#@font = Gosu::Font.new(self, 'assets/fonts/Roboto/Roboto-Regular.ttf', 20)
```

But I still get, each time, the same error :
```
ruby ./test.rb
Mar 21 17:12:04 XXXX.local ruby[45635] <Error>: missing or invalid scheme.
./test.rb:13:in `draw': Cannot load default font (RuntimeError)
	from ./test.rb:13:in `draw'
	from /Users/mickael/.rvm/gems/ruby-2.4.0/gems/gosu-0.11.2/lib/gosu/patches.rb:152:in `tick'
	from /Users/mickael/.rvm/gems/ruby-2.4.0/gems/gosu-0.11.2/lib/gosu/patches.rb:152:in `tick'
	from ./test.rb:17:in `<main>'
```

These fonts are TTF fonts, as specified in [the documentation](http://www.rubydoc.info/github/gosu/gosu/Gosu%2FFont:initialize), and are downloadable from google fonts here:
 - https://fonts.google.com/specimen/Lato
  - https://fonts.google.com/specimen/Roboto

**Comment by jlnr** [comment]
When Gosu cannot load a font file, it tries to load the default font instead (Arial). Since that doesn't work, you (only) see the error message about the default font.

I know this sounds unlikely, but can you please try to install Gosu without rvm (e.g. using system Ruby)? I don't trust rvm's automatic dependency installation at all.

**Comment by Micka33** [comment]
Hi @jlnr,

I see a 2 issues in my previous post:
1. It fails to load the google fonts (Lato&Roboto). `Mar 21 17:12:04 XXXX.local ruby[45635] <Error>: missing or invalid scheme.`
2. It fails to load the default font. `Cannot load default font (RuntimeError)`

I'll try your suggestion about issue 2.
Do you have any idea about issue 1 ?

**Comment by Micka33** [comment]
@jlnr 
With my system ruby

```
 ruby -v
ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
 which ruby
/usr/bin/ruby
 ruby test.rb
Mar 22 18:45:08 XXXX.local ruby[39253] <Error>: missing or invalid scheme.
test.rb:13:in `draw': Cannot load default font (RuntimeError)
  from test.rb:13:in `draw'
  from /Library/Ruby/Gems/2.0.0/gems/gosu-0.11.2/lib/gosu/patches.rb:152:in `tick'
  from /Library/Ruby/Gems/2.0.0/gems/gosu-0.11.2/lib/gosu/patches.rb:152:in `tick'
  from test.rb:17:in `<main>'
```

**Comment by jlnr** [comment]
@Micka33 Hmm. Are `Arial.ttf` and `Arial Bold.ttf` in `/Library/Fonts`? Or have they been disabled? I'm not sure if I should protect against that case as.

**Comment by jlnr** [comment]
Issue 1, the "missing or invalid scheme" bug, looks like a refactoring issue that must have happened in Gosu 0.11.0, sorry about that.

I've pushed a preview gem with a fix. Can you please `gem install gosu --pre` and retry?

**Comment by jlnr** [comment]
No feedback about Arial.ttf -> closing this. The main issue has been fixed in 0.11.3/0.12.0 anyway.
